src/parsers/i3d/main.ts

```diff
@@ -108,6 +108,23 @@ function unpackData(
     numMergedMeshes: 0,
   };
   unpackPrimitives(rootSector, uncompressedValues, compressedData, maps, filterOptions);
+  const sectors = maps.idToSectorMap;
+  const numberOfTreeIndices = Math.max(maps.treeIndexNodeIdMap.length, findLargestTreeIndexInMeshes(compressedData));
+  const loadMeshesCallback = () => { return new Promise(function(resolve, reject) {
+    parseMerged(rootSector, uncompressedValues, compressedData, maps, sceneStats).then(() => {
```

Comment:
Do we need to create a new promise to call resolve on?

Reply:
I think so? unpackData is not async, so it doesn't let me call await.